com.chartboost.mediation/Android~/chartboost-mediation-android-bridge/src/main/java/com/chartboost/mediation/unity/EventProcessor.kt

```diff
@@ -13,7 +15,35 @@ object EventProcessor {
         = eventConsumer.accept(placementName)
 
     @JvmStatic
-    fun serializeEventWithError(placementName: String, error: ChartboostMediationAdException?, eventConsumer: EventWithErrorConsumer<String, String>)
+    fun serializeLoadEvent(
+        placementName: String, loadId: String?, data: Map<String, String>?, error: ChartboostMediationError?,
+        loadConsumer: LoadEventConsumer<String, String, String, String, Double, String>) {
+        val errorMessage = error?.toString() ?: ""
+        var partnerId = ""
+        var auctionId = ""
+        var price =  0.0
+        val adLoadId = loadId ?: ""
+
+        data?.let { winningBidInfo ->
+            partnerId = winningBidInfo["partner_id"] ?: ""
+            auctionId = winningBidInfo["auction-id"] ?: ""
+            price = try {
+                winningBidInfo["price"]?.toDouble() ?: 0.0
+            } catch (e: NumberFormatException) {
+                Log.d(TAG, "HeliumBidEvent failed to serialize price, defaulting to 0.0", e)
+                0.0
+            }
+        }
+
+        loadConsumer.accept(placementName, adLoadId, auctionId, partnerId, price, errorMessage)
+    }
+
+    @JvmStatic
+    fun serializeEventWithError(placementName: String, error: ChartboostMediationError?, eventConsumer: EventWithErrorConsumer<String, String>)
+        = eventConsumer.accept(placementName,error?.toString() ?: "")
+
+    @JvmStatic
+    fun serializeEventWithException(placementName: String, error: ChartboostMediationAdException?, eventConsumer: EventWithErrorConsumer<String, String>)
         = eventConsumer.accept(placementName,error?.toString() ?: "")
```

Comment:
We might not need both of these since `ChartboostMediationAdException` wraps `ChartboostMediationError` so you can extract the latter from the former.

Reply:
Actually, one of them was a duplicate that was not supposed to be there since I took a different approach for Fullscreen API. Thanks!